system/processes/variables.tf

```diff
@@ -0,0 +1,106 @@
+# Global
+
+variable "environment" {
+  description = "Infrastructure environment"
+  type        = string
+}
+
+# SignalFx Module specific
+
+variable "notifications" {
+  description = "Notification recipients list for every detectors"
+  type        = list
+}
+
+variable "prefixes" {
+  description = "Prefixes list to prepend between brackets on every monitors names before environment"
+  type        = list
+  default     = []
+}
+
+variable "filter_custom_includes" {
+  description = "List of tags to include when custom filtering is used"
+  type        = list
+  default     = []
+}
+
+variable "filter_custom_excludes" {
+  description = "List of tags to exclude when custom filtering is used"
+  type        = list
+  default     = []
+}
+
+variable "detectors_disabled" {
+  description = "Disable all detectors in this module"
+  type        = bool
+  default     = false
+}
+
+#####
+
+variable "processes_disabled" {
+  description = "Disable all alerting rules for processes detector"
+  type        = bool
+  default     = null
+}
+
+variable "processes_disabled_critical" {
+  description = "Disable critical alerting rule for processes detector"
+  type        = bool
+  default     = null
+}
+
+variable "processes_disabled_warning" {
+  description = "Disable warning alerting rule for processes detector"
+  type        = bool
+  default     = null
+}
+
+variable "processes_notifications" {
+  description = "Notification recipients list for every alerting rules of processes detector"
+  type        = list
+  default     = []
+}
+
+variable "processes_notifications_warning" {
+  description = "Notification recipients list for warning alerting rule of processes detector"
+  type        = list
+  default     = []
+}
+
+variable "processes_notifications_critical" {
+  description = "Notification recipients list for critical alerting rule of processes detector"
+  type        = list
+  default     = []
+}
+
+variable "processes_aggregation_function" {
+  description = "Aggregation function and group by for processes detector (i.e. \".mean(by=['host']).\")"
+  type        = string
+  default     = ""
+}
+
+variable "processes_transformation_function" {
+  description = "Transformation function for processes detector (mean, min, max)"
+  type        = string
+  default     = "min"
+}
+
+variable "processes_transformation_window" {
+  description = "Transformation window for processes detector (i.e. 5m, 20m, 1h, 1d)"
+  type        = string
+  default     = "1h"
+}
+
+variable "processes_threshold_critical" {
+  description = "Critical threshold for processes detector"
+  type        = number
+  default     = 1
+}
+
+variable "processes_threshold_warning" {
+  description = "Warning threshold for processes detector"
+  type        = number
+  default     = 1
```

Comment:
I am not sure how you think the behavior of this detector but it seems not natural to have same threshold for warning and critical.
I think we should choose a default behavior (like critical if no process at all) and may be make it possible to change behavior to another (like warning if the number of processes do not match).
But the default one should be generic and "plug and play". 
It is possible to add into "Notes" of readme if there is a "custom way" to use.

Reply:
hmmm, so, at first I was not thinking about adding a warning, it will not be used very often, only if the customer need to check one of his own process and need to have at least x processes running.
By default, we will use this detector, with warning in disable state, to monitor processes like sshd, cron, rsyslogd.
So I think I have to put this:

```
53 variable "processes_disabled_warning" {
 54   description = "Disable warning alerting rule for processes detector"
 55   type        = bool
 56   default     = null
 57 }
```
like:

`56   default     = true`

We are speaking about our experience in ex-oxalide, if you see others implementations of this detector, feel free to discuss it with me.

That’s why we used "collectd processes" and not "telegraf processstat" that, for me, is too verbose and with too many options not needed.